pkg/util/priority/priority.go

```diff
@@ -36,32 +37,60 @@ func Priority(w *kueue.Workload) int32 {
 	return ptr.Deref(w.Spec.Priority, constants.DefaultPriority)
 }
 
+// GetPriorityFromPriorityClassAndSource returns the priority from priorityClassName and source.
+func GetPriorityFromPriorityClassAndSource(ctx context.Context, client client.Client, priorityClassName string, source string) (int32, error) {
+	if source == constants.WorkloadPriorityClassSource {
+		_, priority, _, err := GetPriorityFromWorkloadPriorityClass(ctx, client, priorityClassName)
+		return priority, err
+	}
+	// priority belongs k8s priorityClass
+	if source == constants.K8sPriorityClassSource {
+		_, priority, _, err := GetPriorityFromPriorityClass(ctx, client, priorityClassName)
+		return priority, err
+	}
+	err := fmt.Errorf("failed to find priorityClass: %s, source: %s", priorityClassName, source)
+	return 0, err
+}
+
 // GetPriorityFromPriorityClass returns the priority populated from
 // priority class. If not specified, priority will be default or
 // zero if there is no default.
 func GetPriorityFromPriorityClass(ctx context.Context, client client.Client,
-	priorityClass string) (string, int32, error) {
+	priorityClass string) (string, int32, string, error) {
 	if len(priorityClass) == 0 {
 		return getDefaultPriority(ctx, client)
 	}
 
 	pc := &schedulingv1.PriorityClass{}
 	if err := client.Get(ctx, types.NamespacedName{Name: priorityClass}, pc); err != nil {
-		return "", 0, err
+		return "", 0, "", err
 	}
 
-	return pc.Name, pc.Value, nil
+	return pc.Name, pc.Value, constants.K8sPriorityClassSource, nil
+}
+
+// GetPriorityFromWorkloadPriorityClass returns the priority populated from
+// workload priority class. If not specified, returns 0.
+// DefaultPriority is not called within this function
+// because k8s priority class should be  checked next.
+func GetPriorityFromWorkloadPriorityClass(ctx context.Context, client client.Client,
```

Comment:
why should this return the source? It's always going to be WorkloadPriorityClassSource

Reply:
Because I thought the case source is ""  when the priorityClass doesn't exist.
But I agree with your opinion.

Reply:
I return source in this function.

Within the GetPriorityFromPriorityClass function, if there is no default priorityClass, it returns a priority of 0. In this case, we want to set souce "".